py-gssapi: Add kerberos5 variant, to allow use of MIT Kerberos

akkornel opened the pull request:
#### Description

Today, the python-gssapi software builds against macOS' built-in GSSAPI/Kerberos Frameworks.  This adds a variant to build against MacPorts' MIT Kerberos package, `kerberos5`.

kerberos5 is added as a variant because, if someone already has the package installed, we don't want to suddenly switch them to MIT Kerberos during an upgrade.

The long description is also updated to reflect which GSSAPI/Kerberos installation is being used.

Finally, a change is made to the options available on Mac OS X below 10.9: The kerberos5 variant is made the default; if the user attempts to remove this variant, their install errors out.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 15.3 24D60 arm64
Command Line Tools 16.2.0.0.1.1733547573

###### Verification
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)? -- not squashed yet, just in case this aids in reviewing.  I'll squash once given the OK to do so.
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vs install`? -- `sudo port -vst install` fails with exit code 9 when un-tarring the software.
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

tobypeterson commented:
This is a harmless enough change, but I think a better one might be to have this always use kerberos5. Would require some minor edits to setup.py.

Thoughts?

akkornel commented:
Hello!  I've been thinking on your question, and I wonder: Can I explore the possibility of making "Use `kerberos5` instead of GSS.Framework" a variant?  That way, if anyone depends on how py-gssapi is built today (for whatever reason), they won't be affected.

Does that sound like a good path to explore?

tobypeterson commented:
That seems reasonable. Variant to build against kerberos5, and make that the default variant for older versions of macOS. Will require a bit of creative patching.

akkornel commented:
Updated the title and description to reflect the change in purpose of the PR: Instead of just modifying the long description, this is now the addition of a new variant.

> That seems reasonable. Variant to build against kerberos5, and make that the default variant for older versions of macOS. Will require a bit of creative patching.

Happily, https://github.com/pythongssapi/python-gssapi/issues/347 describes a way to build against MIT Kerberos, without needing to patch `setup.py`.  Indeed, this change means that a `setup.py` patch is not needed anymore (at least, from what I can tell).

Updated port passes `port lint --nitpick` and `sudo port test`, at least on my system.  I have not squashed yet, but will do so when given the OK.

Unfortunately, I do not have an older Mac OS X system on which to test those changes to the Portfile.  Is there a place I can go to test?

tobypeterson commented:
Thanks for finding that PR and updating the change!

I don't know of an easy way to test on old systems. The patching for pre-10.9 was done in response to https://trac.macports.org/ticket/60729 - @ryandesign are you able to try this change? It seems like it should work...

tobypeterson commented:
Merging this. Will see if there are any complaints!
